pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/GrpcSendingMailbox.java

```diff
@@ -67,28 +66,22 @@ public class GrpcSendingMailbox implements SendingMailbox {
   private final long _deadlineMs;
   private final StatMap<MailboxSendOperator.StatKey> _statMap;
   private final MailboxStatusObserver _statusObserver = new MailboxStatusObserver();
-  private final boolean _splitBlocks;
   private final int _maxByteStringSize;
+  private final Sender _sender;
 
   private StreamObserver<MailboxContent> _contentObserver;
 
   public GrpcSendingMailbox(
       PinotConfiguration config, String id, ChannelManager channelManager, String hostname, int port, long deadlineMs,
-      StatMap<MailboxSendOperator.StatKey> statMap) {
+      StatMap<MailboxSendOperator.StatKey> statMap, boolean splitBlocks, int maxByteStringSize) {
```

Comment:
IICU `splitBlocks` will be true if and only if `maxByteStringSize` > 0. I would recommend to use a single attribute here.

Reply:
Yup, just used `splitBlocks` locally and assigning `maxByteStringSize=0` if we are not splitting them.